File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {

        boolean isDuplicate = false;
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
            if (map.get(num) == 2) {
                isDuplicate = true;
                break;
            }
        }
        return isDuplicate;

    }
}

File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
    bool isduplicate = false;
    int n = nums.size();

    unordered_map<int, int> map;

    for (int num : nums) {
        map[num] = map[num] + 1;

        if (map[num] == 2) {
            isduplicate = true;
        }
    }

    return isduplicate;
}
};